realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/utils/UserFactory.java

```diff
@@ -16,14 +16,38 @@
 
 package io.realm.objectserver.utils;
 
+import java.math.BigInteger;
+import java.security.SecureRandom;
+
+import io.realm.Realm;
+import io.realm.RealmConfiguration;
 import io.realm.SyncCredentials;
 import io.realm.SyncUser;
+import io.realm.log.RealmLog;
 
 // Must be in `io.realm.objectserver` to work around package protected methods.
 public class UserFactory {
+    private static final String PASSWORD = "myPassw0rd";
+    // Since the integration tests need use the same user for different processes, we create a new user name when the
+    // test starts and store it in a Realm. Then it can be retrieved for every process.
+    private String userName;
+    private static UserFactory instance;
+    private static RealmConfiguration configuration = new RealmConfiguration.Builder()
```

Comment:
This require `Realm.init()` to be called before using this class.  Perhaps we should add a comment to the header about that?